detectable/src/main/java/com/synopsys/integration/detectable/detectables/setuptools/parse/SetupToolsCfgParser.java

```diff
@@ -0,0 +1,142 @@
+package com.synopsys.integration.detectable.detectables.setuptools.parse;
+
+import java.io.BufferedReader;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
+
+import org.tomlj.TomlParseResult;
+
+import com.synopsys.integration.detectable.python.util.PythonDependency;
+import com.synopsys.integration.detectable.python.util.PythonDependencyTransformer;
+
+public class SetupToolsCfgParser implements SetupToolsParser {
+    
+    private TomlParseResult parsedToml;
+    
+    private String projectName;
+    
+    private List<String> dependencies;
+    
+    public SetupToolsCfgParser(TomlParseResult parsedToml) {
+        this.parsedToml = parsedToml;
+        this.dependencies = new ArrayList<>();
+    }
+
+    @Override
+    public SetupToolsParsedResult parse() throws IOException {
+        String tomlProjectName = parsedToml.getString("project.name");
+        String projectVersion = parsedToml.getString("project.version");
+        
+        // If we have multiple project names the name from the toml wins
+        // I've only seen version information in the toml so use that.
+        String finalProjectName = (tomlProjectName != null && !tomlProjectName.isEmpty()) ? tomlProjectName : projectName;
+        
+        List<PythonDependency> parsedDirectDependencies = parseDirectDependencies();
+        
+        return new SetupToolsParsedResult(finalProjectName, projectVersion, parsedDirectDependencies);
+    }
+
+    /**
+     * Extracts, does not parse, any entries in the install_requires section of the
+     * setup.cfg
+     * 
+     * @param filePath path to the setup.cfg file
+     * @return a list of dependencies extracted from the install_requires section
+     * @throws FileNotFoundException
+     * @throws IOException
+     */
+    public List<String> load(String filePath) throws FileNotFoundException, IOException {
+        try (BufferedReader reader = new BufferedReader(new FileReader(filePath))) {
+            String line;
+
+            // This flag is used to indicate whether we are currently reading the lines under the "install_requires" key
+            boolean isInstallRequiresSection = false;
+
+            while ((line = reader.readLine()) != null) {
+                line = line.trim();
+                
+                if (line.startsWith("name")) {
+                    parseProjectName(line);
+                }
+
+                // Remove all whitespace from the line for key searching
+                String keySearch = line.replaceAll("\\s", "");
```

Comment:
Can project names have whitespaces?

Reply:
I believe so, I'm only trimming the beginning of the line. I don't strip all the whitespace until after the name is extracted and we are looking for the install_requires= portion.